package/kedro_viz/server.py

```diff
@@ -488,10 +488,42 @@ def _get_dataset_node(node_id, namespace):
 
 @app.route("/api/main")
 def nodes_json():
-    """Serve the pipeline data. This includes basic node data amongst others edges, tags, and layers."""
+    """Serve the data from all Kedro pipelines in the project.
+    This includes basic node data amongst others edges, tags, and layers.
+    """
     return jsonify(_DATA)
 
 
+@app.route("/api/pipelines/<string:pipeline_id>")
+def pipeline_data(pipeline_id):
+    """Serve the data from a single pipeline in a Kedro project."""
+    current_pipeline = {"id": pipeline_id, "name": _pretty_name(pipeline_id)}
+    pipelines_list = _DATA["pipelines"]
+    pipelines_list.remove(current_pipeline)
+    pipelines_list.insert(0, current_pipeline)
```

Comment:
Why do we need to move the pipeline in the first elemennt in the list?

Reply:
It's in the ticket. The FE takes the first element as the selected pipeline.

Reply:
Can't the FE store the selected pipeline in the local storage? Relying on the order in thee list is very hacky to me.

Reply:
Actually we do this in https://github.com/quantumblacklabs/kedro-viz/blob/main/package/kedro_viz/server.py#L374-L375 as well .. 
@limdauto I remember that you suggested adding a new entry `selected_json` the JSON a while ago instead of using the first index in the list. Maybe something we could implement as a follow-up PR?

Reply:
I did propose adding a new field called `selected_pipeline`. Can't remember why it was rejected. Let me check with Richard again.

Reply:
I don't think it was rejected. I remember he was happy with it?

Reply:
1. Yeah I'm happy either way - you can either use the first pipeline in the list to indicate the default one (i.e. the one that should display when you start the server), or you can set an extra prop with that value. I just figured that using the first one is easier and avoids having an extra prop, but maybe it's good to be explicit about it. Don't mind.
2. We actually are already using localStorage to save the value of the last pipeline the user had open and reuse it where possible. However we need to know which pipeline to display in cases where the user does not already have anything stored in localStorage, or in cases where the stored value does not match any of the existing pipelines. By default the front-end will just serve up the first pipeline in the list - see point 1.

Reply:
👍 thanks @richardwestenra I will add another prop then.